os/net/lwip/src/core/ipv6/nd6.c

```diff
@@ -428,27 +428,15 @@ void nd6_input(struct pbuf *p, struct netif *inp)
 
 		/* Check if there is a link-layer address provided. Only point to it if in this buffer. */
 		if (p->len >= (sizeof(struct ns_header) + 2)) {
-			u32_t offset = sizeof(struct ns_header);
-
-			while (p->len >= offset + 2) {
-				lladdr_opt = (struct lladdr_option *)((u8_t *) p->payload + offset);
-				if (ND6H_LLADDR_OPT_LEN(lladdr_opt) == 0) {
-					pbuf_free(p);
-					ND6_STATS_INC(nd6.lenerr);
-					ND6_STATS_INC(nd6.drop);
-					return;
-				}
-
-				if (ND6H_LLADDR_OPT_TYPE(lladdr_opt) != ND6_OPTION_TYPE_SOURCE_LLADDR) {
-					/* RFC 4861 clause 4.3
-					 * only source lladdr option is possible option for NS message */
-					lladdr_opt = NULL;
-				}
-				if (p->len < (offset + (ND6H_LLADDR_OPT_LEN(lladdr_opt) << 3))) {
-					lladdr_opt = NULL;
-				}
-
-				offset += 2;
+			lladdr_opt = (struct lladdr_option *)((u8_t *) p->payload + sizeof(struct ns_header));
```

Comment:
We don't need to merge "merge and revert". Could you remove both, 0f5fc9ee5175a9782abd2647102ca714169cfa29 and this?